File: 241003/사각형의 총 넓이 2/total-width-of-a-rectangle2.py
N = int(input())
# -100 ~ 100
arr = [[0]*201 for i in range(201)]
for i in range(N):
    x1, y1, x2, y2 = map(int, input().split())
    for m in range(x1, x2):
        for n in range(y1, y2):
            arr[m+100][n+100] = 1

cnt = 0
for i in range(201):
    for j in range(201):
        if arr[i][j] == 1:
            cnt += 1
print(cnt)

File: 250106/사각형의 총 넓이 2/total-width-of-a-rectangle2.cpp
#include <iostream>
using namespace std;

int main() {
    // Please write your code here.
    int N, arr[201][201];
    cin >> N;
    for (int i=0; i<N; i++) {
        int x1, y1, x2, y2;
        cin >> x1 >> y1 >> x2 >> y2;
        for (int x=x1; x<x2; x++) {
            for (int y=y1; y<y2; y++) {
                arr[x][y] = 1;
            }
        }
    }
    int cnt = 0;
    for (int i=0; i<201; i++) {
        for (int j=0; j<201; j++) {
            if (arr[i][j] == 1) cnt++;
        }
    }
    cout << cnt;
    return 0;
}